src/NetworkLayout.jl

```diff
@@ -1,11 +1,96 @@
 module NetworkLayout
 
+using GeometryBasics
+using Requires
+using LinearAlgebra: norm
+using Random
+
+export LayoutIterator, layout
+
+"""
+    AbstractLayout{Dim,Ptype}
+
```

Comment:
It would be good to document what the parameters `Dim` and `Pyype` are. and what types the can take.

Also thinking about it, how often is the case, that `Dim` will not be `2`  and `Ptype` not be `Float64`? Are they really that important that always specify them in the type? Other possibilities would be to specifying these parameters when calling the `layout`  function, maybe as `kwargs` or just have the default trait functions defined like this
```
dim(::AbstractLayout) = 2
ptype(::AbstractLayout) = Float64
```
then one can override these trait functions for a concrete layout, if they would differ somehow,

Reply:
The functions `dim` and `ptype` aren't used at all in the moment tbh, i am not sure whether they are important at all. But the parameters are used in the `layout` functions quite often.

- `Dim`: Right now there are 3 layouts which support 2D and 3D (Spring¹, Stress, SFDP), one pure 3D layout (Spectral) and 3 pure 2D layouts (Buchheim, Circular/Shell², SquareGrid). Each layout has the intrinsic property of a dimensionality, I think this justifies to put this information to the supertype. Otherwise we need to add an explicit `dim` type parameter to 3 of the layouts (should be a type parameter rather than a field becaus it determines the return value `Vector{Point{Dim,Ptype}}` of the `layout` function.
- `Ptype`: Well, the number type of the return values was parametrized before and I did not change that. I'm not gonna lie: this whole interface is opinionated and developed with GraphMakie in mind. And since Makie uses the GPU it is quit common to work with `Float32` too. 
  
¹ there is a minor parametrization bug in the 3d spring implementation though but I did not consider this part of the interface
² We don't actually need special code for `Circular`. The resulting layout of `Circular()` and `Shell()` without parameters is the same.